src/pdf_beamtime/src/pick_place_server.cpp

```diff
@@ -0,0 +1,158 @@
+/*Copyright 2023 Brookhaven National Laboratory
+BSD 3 Clause License. See LICENSE.txt for details.*/
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <functional>
+#include <memory>
+#include <thread>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_action/rclcpp_action.hpp>
+#include <rclcpp_components/register_node_macro.hpp>
+#include <geometry_msgs/msg/pose.hpp>
+#include <moveit/planning_scene_interface/planning_scene_interface.h>
+#include <moveit/move_group_interface/move_group_interface.h>
+#include <yaml-cpp/yaml.h>
+#include <hello_moveit_interfaces/action/pick_place_repeat.hpp>
+
+using namespace std::chrono_literals;
+
+// This only adds the obstacle environment. This should be changed to have a header in the future iterations
+
+class PickPlaceServer
+{
+public:
+  using PickPlaceRepeat = hello_moveit_interfaces::action::PickPlaceRepeat;
```

Comment:
make your own action and interfaces. We can iterate on these, but it's an important part of the process.